### ISSUE TITLE
Can't merge next into master when next is two versions ahead

### ISSUE THREAD

**Issue by jamime** [semantic-release/semantic-release#1398]
Firstly, thank you! I'm really grateful for `semantic-release` 🎉 
## Current behavior
I've been getting familiar with the process defined in https://github.com/semantic-release/semantic-release/blob/e5344cd89bc47a3a8608f0d62353cd9c660a1bd9/docs/recipes/distribution-channels.md

What is the expected way to merge `next` into `master`? In my example I have

- `master` (v4.0.0)
- `next` (v5.0.0)

I want to make the features in next available on the `@latest` dist channel but when I release it says
> Based on the releases published on other branches, only versions within the range >=4.0.0 <5.0.0 can be published from branch master.

## Expected behavior

I expect to do a git merge to make `next` available on the `@latest` dist channel. 

## Environment

- **semantic-release** version: 16.0.0-beta.46
- CI environment: GH Actions
- Plugins used: See below
- **semantic-release** configuration: 
```
"release": {
    "preset": "conventionalcommits",
    "plugins": [
      "@semantic-release/commit-analyzer",
      "@semantic-release/release-notes-generator",
      "@semantic-release/changelog",
      "@semantic-release/npm",
      "@semantic-release/git",
      "@semantic-release/github"
    ]
  }
```
- CI logs: https://github.com/jamime/semantic-example-two/commit/30b4905bfd149fb249ff2d8bdd85ef4ed389bf07/checks?check_suite_id=367203387

**Comment by jamime** [comment]
It does work when `next` is one version ahead. I wonder if it was because `master` was two version behind. Trying this theory now.

**Comment by jamime** [comment]
Confirmed.

Steps to reproduce

- Release version `2.0.0` from `master`
- Release version `3.0.0` from `next`
- Release version `4.0.0` from `next`
- Release version `2.0.1` from `master`
- Merge `next` into `master`

https://github.com/jamime/semantic-example-two/issues/1
https://github.com/jamime/semantic-example-two/commit/30b4905bfd149fb249ff2d8bdd85ef4ed389bf07/checks?check_suite_id=367203387

**Comment by jamime** [comment]
Also, once you are in this situation I don't think you can get out because you can not make major releases from `master` anymore in this case because the `3.0.0` tag will already exist on `next`.

**Comment by pvdlg** [comment]
Looking at your repo `semantic-example-two` I see that the tags on `master` are `v1.0.0`, `v2.0.0`,  `v2.0.1`, `v3.0.0` and on `next` are `v1.0.0`, `v2.0.0`,  `v3.0.0`, `v4.0.0`.

That doesn't seems to match what you described in  https://github.com/semantic-release/semantic-release/issues/1398#issuecomment-567608824 where you released `v4.0.0` on `next` before merging `next` into `master`. `v4.0.0` should be in the history of `master` if `next` was merged into `master` after releasing `v4.0.0` on `next`.

The error you get in repo `semantic-example-two` make sense because the last release on `master` is `3.0.0`, while the last one on `next` is `4.0.0` (i.e. `4.0.0` was not merged to master). So if you make a breaking change on master that would end up releasing `4.0.0` which has already been released on `next`. Before making a breaking change on `master` you must merge `4.0.0` from next to `master`, otherwise this breaking has to be made on `next` as indicated in the error message.

It seems you think you merged `4.0.0` into master, but it's not there (as shown by `git tag --merged master`). Could you clarify?

**Comment by jamime** [comment]
🤦‍♂ this was user error. Because of the plugins that I have configured it commits to the branch to update the changelog and package.json. That remote commit is the tag that is released, however, I merged my local `next` branch which did not have the tag associated.

Sorry for wasting your time.
